app/scripts/components/common/map/controls/aoi/index.tsx

```diff
@@ -47,33 +52,38 @@ export default function DrawControl(props: DrawControlProps) {
         },
         ...props
       });
-      return control.current;
+      return control;
     },
-    ({ map }: { map: any }) => {
-      map._drawControl = control.current;
+    ({ map }) => {
+      mapRef.current = map as ExtendedMapRef;
+      const extendedMap = map as ExtendedMapRef;
```

Comment:
Any specific reason you decide to use object reference here? I tend to lean towards not using object reference (I think it doesn't help readability and makes it challenging to see where the states are mutated), but I would like to hear your thoughts behind the decision!

Reply:
Yeah, I was wondering if ts was the reason, and I think your workaround should work 👍